corehq/apps/app_manager/views/download.py

```diff
@@ -368,12 +383,36 @@ def validate_form_for_build(request, domain, app_id, unique_form_id, ajax=True):
         return HttpResponse(response_html)
 
 
-def download_index_files(app):
+def _file_needed_for_CCZ(path, prefix, profiles):
+    if path.startswith(prefix):
+        # already narrowed to specific profile
+        if prefix != 'files/':
+            return True
+        else:
+            second = path.split('/')[1]
+            # dont want to return files from other profiles
+            if second in profiles:
+                return False
+            else:
+                return True
```

Comment:
the above suggestion should be inverted (`return second not in profiles`). But I think the logic would be clearer if this were converted to lambda functions inside `download_index_files`:
```py
def download_index_files(app, build_profile_id=None):
    if app.copy_of:
        prefix = 'files/'
        if build_profile_id:
            prefix += build_profile_id + '/'
            needed_for_CCZ = lambda path: path.startswith(prefix)
        else:
            profiles = set(app.build_profiles)
            needed_for_CCZ = lambda path: path.split('/')[1] not in profiles
        if not (prefix + 'profile.ccpr') in app.blobs:
            # profile hasnt been built yet
            app.create_build_files(save=True, build_profile_id=build_profile_id)
            app.save()
        files = [(path[len(prefix):], app.fetch_attachment(path))
                 for path in app.blobs if needed_for_CCZ(path)]
    else:
        files = app.create_all_files().items()
    return sorted(files)
```
That way there are less logical branches and variables to keep track of when reading the code.